src/components/Clock.tsx

```diff
@@ -0,0 +1,49 @@
+import React from 'react';
+
+type Props = {
+  clockName: number,
+};
+
+type State = {
+  date: string;
+};
+
+export class Clock extends React.Component<Props, State> {
+  state = {
+    date: new Date().toLocaleTimeString(),
+  };
+
+  timerId: NodeJS.Timer | undefined;
+
+  componentDidMount() {
+    this.timerId = setInterval(() => {
+      this.setState({ date: new Date().toLocaleTimeString() });
+
+      // eslint-disable-next-line no-console
+      console.log(this.state.date);
+    }, 1000);
+  }
+
+  componentDidUpdate(prevName: Props) {
+    if (this.props.clockName !== prevName.clockName) {
+      // eslint-disable-next-line no-console
+      console.log(`The Clock was renamed from ${prevName.clockName} to ${this.props.clockName}`);
+    }
+  }
```

Comment:
Check if the message appears in the console if you hide clock time
(When I hide clock time I can change the clock name, but I don't see this message on the console)